src/AsmResolver.DotNet/Code/Cil/CilMethodBody.cs

```diff
@@ -13,6 +15,256 @@ namespace AsmResolver.DotNet.Code.Cil
     /// </summary>
     public class CilMethodBody : MethodBody, ICilOperandResolver
     {
+         /// <summary>
+        ///     Creates a CIL method body from a dynamic method.
+        /// </summary>
+        /// <param name="method">The method that owns the method body.</param>
+        /// <param name="dynamicMethodObj">The dynamic method.</param>
+        /// <param name="operandResolver">
+        ///     The object instance to use for resolving operands of an instruction in the
+        ///     method body.
+        /// </param>
+        /// <returns>The method body.</returns>
+        public static CilMethodBody FromDynamicMethod(MethodDefinition method, object dynamicMethodObj,

```

Comment:
This method is too large (too many lines and does too many things).
Defining `dynamicMethodObj` to be `object` is too arbitrary. It is not clear what to plug in here. Maybe define overloads of this method for all supported value types?